tincr/io/device/xdlrc.tcl

```diff
@@ -401,32 +401,59 @@ proc ::tincr::write_primitive_defs { args } {
     }
 }
 
-## Produce a partial (everything but connections) .def file for the given primitive site. All config strings will be put inside of bel elements.
+## Produce a partial .def file for the given primitive site. All config strings will be put inside of bel elements.
+#   For Single Bel Sites (Sites where one BEL has over 80% of all bel pins in the site), some connections will be
+#   automatically inferred and generated.
+#
 # @param site The <CODE>site</CODE> object you want to create a .def file for. Sites instanced by alternate types are acceptable.
 # @param filename The output file.
 # @param includeConfigs A boolean telling the proc whether or not to include configuration elements in the resulting .def file.
 proc ::tincr::write_partial_primitive_def { site filename {includeConfigs 0} } {
     set site [get_sites $site]
     set site_name [get_property NAME $site]
-    puts $filename
     set outfile   [open $filename w]
     set alternate 0
     
-    # **************************************************************************
-    # *                                                                        *
-    # * Pin Resources:                                                         *
-    # *                                                                        *
-    # **************************************************************************
     set site_pins [get_site_pins -of $site -quiet]
     set bels [get_bels -of $site -quiet]
     set site_pin_names [list]
     set num_elements [ expr { [llength $site_pins] + [llength $bels] } ]
+        
+    # Look for "Single Bel Sites" and try to generate as many connections as possible automatically
+    set is_single_bel_site 0
+    set pin_maps [list]
+    if {[llength $bels] == 1} {
+        set is_single_bel_site 1
+        set pin_maps [get_single_bel_pin_maps $site [lindex $bels 0] $outfile]
+    } else {
+        set num_bel_pins_site [llength [get_bel_pins -of $site -quiet]]
+        
+        foreach bel $bels {
+            set num_bel_pins_bel [llength [get_bel_pins -of $bel -quiet]]
+            
+            # if a single bel in a site contains 80% of all bel pins in the site, mark it as a single bel site
+            if {[expr {double($num_bel_pins_bel) / double($num_bel_pins_site)}] > .800} {
```

Comment:
GTHE3_COMMON sites have a bel that constitutes ~85% of bel pins. I wanted to set the cutoff as high as possible but still capture the sites I can generate connections automatically for. Read my other comment for a more thorough explanation.